components/net/tests/main.rs

```diff
@@ -124,7 +124,7 @@ fn new_fetch_context(
     let sender = fc.unwrap_or_else(|| create_embedder_proxy());
 
     FetchContext {
-        state: Arc::new(create_http_state()),
+        state: Arc::new(create_http_state(Some(sender.clone()))),
```

Comment:
Sorry for missing this previously, but since the embedder proxy is already part of FetchContext, why do we need to add it to HttpState? The code in http_loader.rs has access to a FetchContext.

Reply:
you didn't miss it, I just added these changes yesterday. I'll look into it

Reply:
I'm guessing we wouldn't need to add to the HttpState. I'll look into it. I was following the directions [laid out here](https://github.com/servo/servo/issues/27439#issue-667841183). So I wasn't aware that `FetchContext` already had access

Reply:
Well, I feel a bit silly now :D

Reply:
So I didn't see anywhere that the `embedder_proxy` is part of the `FetchContext` besides within the `FileManager`. Unless you want to make the `embedder_proxy` part of the `FetchContext` instead of `HttpState`? I think that might make more sense conceptually. I haven't looked into how much effort it would be.

Reply:
You're right! I'm not sure how I misread the code.